p2p/src/authenticated/actors/tracker/ingress.rs

```diff
@@ -148,6 +156,28 @@ impl<E: Spawner + Metrics, C: Verifier> Oracle<E, C> {
     }
 }
 
+/// Mechanism to block peers in the network.
+#[derive(Clone)]
+pub struct Blocker<E: Spawner + Metrics, C: Verifier> {
+    sender: mpsc::Sender<Message<E, C>>,
+}
+
+impl<E: Spawner + Metrics, C: Verifier> Blocker<E, C> {
+    /// Create a new instance of a blocker.
+    pub(super) fn new(sender: mpsc::Sender<Message<E, C>>) -> Self {
+        Self { sender }
+    }
+}
+
+impl<E: Spawner + Metrics, C: Verifier> crate::Blocker for Blocker<E, C> {
+    type PublicKey = C::PublicKey;
+
+    /// Block a peer, disconnecting them if currently connected and preventing future connections.
```

Comment:
nit: "preventing future connections as long as they are in some active set." (I think we should be very clear about the expectations there)